packages/@lwc/engine-core/src/libs/mutation-tracker/index.ts

```diff
@@ -101,14 +101,23 @@ export class ReactiveObserver {
         if (len > 0) {
             for (let i = 0; i < len; i++) {
                 const set = listeners[i];
-                if (set.length === 1) {
-                    // Perf optimization for the common case - the length is usually 1, so avoid the indexOf+splice.
-                    // If the length is 1, we can also be sure that `this` is the first item in the array.
-                    set.length = 0;
+                // jtu-todo: use the .call annotation here instead
+                if (Array.isArray(set)) {
+                    if (set.length === 1) {
+                        // Perf optimization for the common case - the length is usually 1, so avoid the indexOf+splice.
+                        // If the length is 1, we can also be sure that `this` is the first item in the array.
+                        set.length = 0;
+                    } else {
+                        // Slow case
+                        const pos = ArrayIndexOf.call(set, this);
+                        ArraySplice.call(set, pos, 1);
+                    }
+                } else if (typeof set === 'function') {
+                    set.call(undefined, this);
```

Comment:
If `set` might not actually be an array, a more descriptive variable name would aid readability here.